### ISSUE TITLE
Sites on which ads are hidden but not captured

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#356]
These need to be addressed (note: this is a good exercise for new developers wishing to understand how ad-parsing works):

nytimes.com
piratebay.org

**Comment by dhowe** [comment]
thepiratebay.org finished with #363

**Comment by cqx931** [comment]
ads on nytimes.com are not captured, because the script generating them has been blocked by EasyPrivacy. Solved by adding dynamicFilteringString to allow the request.https://github.com/dhowe/AdNauseam/issues/399

**Comment by cqx931** [comment]
After implementing dynamic FilterString AdNauseam can collect ads images from iframes with external src(in this case, iframe[src*="googlesyndication.com"]). 
However, for the iframes whose src looks like this:
![screen shot 2016-09-20 at 5 07 28 pm](https://cloud.githubusercontent.com/assets/3056154/18664329/c2c5025a-7f54-11e6-9f33-76b4106a60f9.png)
, I can only see the process message of the iframe, but not any content inside.
Could it be some problems with injecting content script into those iframes?

**Comment by cqx931** [comment]
yes, this issue still exists.

**Comment by dhowe** [comment]
can you test whether our content scripts are being (dynamically) injected?

**Comment by cqx931** [comment]
I can see a few successful injection here but I don't know which iframe they are. How can I know the exact element from the frameId?
![screen shot 2016-09-20 at 9 00 39 pm](https://cloud.githubusercontent.com/assets/3056154/18671154/8da24530-7f75-11e6-8b8d-648aede8a633.png)

**Comment by dhowe** [comment]
I'm not exactly sure, as you've guessed those last 2 numbers are tabId / frameId, the URL is for the page itself -- you can also print out the URL from the frame, but there isn't always one

**Comment by cqx931** [comment]
Then the content scripts are only (dynamically) injected into those iframes with an external src, not to those with src"javascript:'<html><body></body></html>'".

**Comment by dhowe** [comment]
Is that true? It should be any iframes that don't exist originally on the page, but are created (regardless of source)

**Comment by cqx931** [comment]
http://www.si.com/
ad iframe won't be dynamically created when the outside ad-container is invisible.
Therefore, no ads can be found when AdNauseam hide all the ad-container. Once the cosmetic filter is toggled on, ads can be collected.

**Comment by dhowe** [comment]
so we need to not block the outer ad-container, correct?
we can do this either by adding the rule to the disabledRules list,
or by creating an exception rule in adnauseam.txt

**Comment by cqx931** [comment]
yes. I think an exception rule for ad-container in si.com would be good, as this is a quite wide selector.

**Comment by dhowe** [comment]
great, make it so..

**Comment by cqx931** [comment]
fixed si.com with https://github.com/dhowe/AdNauseam/pull/433

**Comment by dhowe** [comment]
👍

**Comment by cqx931** [comment]
- [ ] 163.com
- [ ] sohu.com
- [ ] sina.com
- [ ] qq.com
- [ ] zol.com.cn
- [ ] baidu.com
- [ ] weibo.com
- [ ] sogou.com

**Comment by leoneckert** [comment]
ENGLISH checklist of ads hidden but not collected
- [ ] si.com
- [ ] cnn.com (2 hidden, 0 collected)

**Comment by cqx931** [comment]
A fresh update of nytimes...At least for today, some of their ad images(Ex:TopLeft) don't have a parent <a> tag. Instead they have this interesting way of writing onclick attribute for their ad images...And when you click it, it doesn't lead you to anywhere...
<img width="794" alt="screen shot 2016-09-27 at 12 54 14 am" src="https://cloud.githubusercontent.com/assets/3056154/18843859/55fc0f94-844d-11e6-91cb-3b4c243536a8.png">

**Comment by dhowe** [comment]
Not sure if I understand what you are saying: when I click, I end up at the ad site (or do you mean when adn clicks?)

![screen shot 2016-09-27 at 1 14 42 am](https://cloud.githubusercontent.com/assets/737638/18844482/fda9fe16-844f-11e6-9436-fdfce8f0785b.png)

**Comment by cqx931** [comment]
Interesting...if I click the ad without any blocker on, I also end up in the ad site.
But when I have AdNauseam on, if seems to go through a quick process of opening a new page and closing it instantly...
So if this is the case...shall we try to parse this onclick?

**Comment by dhowe** [comment]
Do you mean that you disable cosmetic filters for the page, then view and click the ads?
If so, I notice that disabling blocking for nytimes, via the dynamic filter:

`nytimes.com * * allow`

fixes the problem.

But thus far I am unable to isolate the specific rule that causes the issue. My guess is that it is one of the following (each showing as blocked in the page console):

```
GET https://cdn.optimizely.com/...
GET https://cm.g.doubleclick.net/...
GET https://cdn.krxd.net/...
GET http://dc8xl0ndzn2cb.cloudfront.net/...
```

**Comment by dhowe** [comment]
Note I've also updated the parseOnClick() code to handle this case, see 54ae3fc7

**Comment by cqx931** [comment]
Question: Do we want text-ad parser for Chinese search engines? I can work on that.

**Comment by dhowe** [comment]
yeah, that would be good, at least for the 2 or 3 most popular...

**Comment by leoneckert** [comment]
- [x] yahoo ads were hidden but not collected because internal link. Fixed in https://github.com/dhowe/AdNauseam/pull/464

**Comment by leoneckert** [comment]
- [ ] http://www.accuweather.com/ there is massive ads all over, which get hidden, but do not show up in parser.js console nor are they collected. Are those the so called dynamically created iframes?

**Comment by dhowe** [comment]
They show up in the parser console for me -- did you enable the flag?

![screen shot 2016-09-30 at 6 32 53 am](https://cloud.githubusercontent.com/assets/737638/18974763/eca3135c-86d7-11e6-9250-9afff9132a62.png)

**Comment by dhowe** [comment]
And when I disable blocking via a firewall rule (below), I see many ads being collected, which tells me that one of the blocking rules is breaking the ad collection (probably because elements we need are never making it to the page).

![screen shot 2016-09-30 at 7 22 24 am](https://cloud.githubusercontent.com/assets/737638/18975746/f8ea5678-86de-11e6-8091-612372abb750.png)

![screen shot 2016-09-30 at 7 22 35 am](https://cloud.githubusercontent.com/assets/737638/18975753/fcd34222-86de-11e6-9da0-c463424fe9c7.png)

**Comment by leoneckert** [comment]
pages analysed in https://github.com/dhowe/AdNauseam/issues/427 in which ads were hidden but not collected. working on this next.

- [ ] https://www.yahoo.com/news
- [ ] http://www.cnn.com
- [ ] http://www.nytimes.com/
- [ ] http://www.nbcnews.com/
- [ ] http://www.dailymail.co.uk/ushome/
- [ ] https://www.washingtonpost.com
- [ ] https://www.theguardian.com/us
- [ ] http://www.wsj.com
- [ ] http://abcnews.go.com
- [ ] http://www.bbc.com/news
- [ ] http://www.usatoday.com
- [ ] http://www.cbsnews.com
- [ ] http://www.reuters.com
- [ ] http://www.msnbc.com
- [ ] http://www.cbc.ca/news
- [ ] http://www.news.com.au

**Comment by leoneckert** [comment]
^ going through above websites. Is there a proven method to definitively say what ads are not collected? What I experience often is that an ad is not collected 5 times in a row and suddenly when refreshing a 6th time it is. Result of that is that I write filters and during testing suddenly notice that it is redundant. 
I am aware events fire irregularly and things might be dependent on cache or if I toggle cosmetic filtering (which might activate dynamic iFrames to be generated) etc. Wondered if you have similar experiences / a rule of thumb for me.

**Comment by dhowe** [comment]
Items below are for >= v3.0

http://www.overclockers.ru  (3 banner ads at top, one left nav ad)

**Comment by dhowe** [comment]
@sindhubr

**Comment by mneunomne** [comment]
Seems that Baidu is currently not collecting ads on search engine, need to implement it since there are so many potential ad collections
